cmd/query/app/server.go

```diff
@@ -109,8 +109,19 @@ func createGRPCServer(querySvc *querysvc.QueryService, metricsQuerySvc querysvc.
 	}
 	if tm.Enabled {
 		grpcOpts = append(grpcOpts,
-			grpc.StreamInterceptor(tenancy.NewGuardingStreamInterceptor(tm)),
-			grpc.UnaryInterceptor(tenancy.NewGuardingUnaryInterceptor(tm)),
+			grpc.ChainUnaryInterceptor(
+				tenancy.NewGuardingUnaryInterceptor(tm),
+				bearertoken.NewUnaryServerInterceptor(),
+			),
```

Comment:
Please do not couple tenancy and new interceptors. Define a slice of interceptors initialized to bearertoken only, then optionally add tenancy inside the if(), and then at the end add the actual grpc option.

Reply:
Got it